apis/python/src/tiledb/vector_search/object_readers/directory_reader.py

```diff
@@ -115,39 +344,107 @@ def get_partitions(
         return partitions
 
 
+class TileDBLoader:
+    def __init__(
+        self,
+        uri: str,
+        config: Optional[Mapping[str, Any]] = None,
```

Comment:
I believe we should also support passing a context so it can be reused.

Reply:
See comment for removing context